File: 프로그래머스/lv2/12911. 다음 큰 숫자/다음 큰 숫자.js
function solution(n) {
    let answer = 0;
    for(let i=n+1; i<=1000000; i++) {
        if(getCount(n.toString(2)) === getCount(i.toString(2))) {
            answer = i;
            break;
        }
    }
    return answer;
}


function getCount(num) {
    let cnt = 0;
    for(let i=0; i<num.length; i++) {
        if(num[i] === '1') cnt++;
    }
    return cnt;
}

File: 프로그래머스/lv2/12911. 다음 큰 숫자/다음 큰 숫자.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(int n) {
        int answer = 0;
        
        for(int i=n+1; i<=1000000; i++) {
            if(getCount(Integer.toString(n,2)) == getCount(Integer.toString(i,2))) {
                answer = i;
                break;
            }
        }
        return answer;
    }
    
    public int getCount(String num) {
        int count = 0;
        for(int i=0; i<num.length(); i++) {
            if(num.charAt(i) == '1') count++;
        }
        return count;
    }
}